deeplabcut/modelzoo/apis/spatiotemporal_adapt.py

```diff
@@ -0,0 +1,196 @@
+
+        Examples
+        --------
+
+        from  deeplabcut.modelzoo.apis import SpatiotemporalAdaptation
+        video_path = '/mnt/md0/shaokai/openfield_video/m3v1mp4.mp4'
+        init_weights = '/mnt/md0/shaokai/DLC-ModelZoo/ma_supertopview/dlc-models/iteration-0/ma_supertopviewMarch30-trainset95shuffle1/train/snapshot-200000'
+
+        videotype = 'mp4'
+        >>> adapter = SpatiotemporalAdaptation(video_path,
+                                       init_weights,
+                                       'supertopview',
+                                       modelfolder = "temp_topview",
+                                       videotype = videotype)
+
+        adapter.before_adapt_inference()
+        adapter.adaptation_training()
+        adapter.after_adapt_inference()
+
+
+        """
+        if supermodel_name not in ["superquadruped", "supertopview"]:
+            raise ValueError("`supermodel_name` should be either 'superquadruped' or 'supertopview'")
+        self.video_path = video_path
+        self.init_weights = init_weights
+        self.supermodel_name = supermodel_name
+        self.scale_list = scale_list
+        self.videotype = videotype
+        vname = str(Path(self.video_path).stem)
+        self.adapt_modelprefix = vname + "_video_adaptation"
+        self.adapt_iterations = adapt_iterations
+        self.modelfolder = modelfolder
+        self.pcutoff = pcutoff
+        self.pseudo_threshold = pseudo_threshold
+
+        if modelfolder != "":
+            os.makedirs(modelfolder, exist_ok=True)
+
+            dlc_root_path = os.sep.join(deeplabcut.__file__.split(os.sep)[:-1])
+
+            name_dict = {
+                "supertopview": "supertopview.yaml",
+                "superquadruped": "superquadruped.yaml",
+            }
+
+            self.customized_pose_config = os.path.join(
+                dlc_root_path,
+                "pose_estimation_tensorflow",
+                "superanimal_configs",
+                name_dict[self.supermodel_name],
+            )
+
+        if customized_pose_config != "":
+            # if it's an old modelzoo model, this is also required
+            self.customized_pose_config = customized_pose_config
+
+    def before_adapt_inference(self):
+
+        # save frames have to be on
+        deeplabcut.video_inference_superanimal(
+            [self.video_path],
+            self.supermodel_name,
+            videotype=self.videotype,
+            scale_list=self.scale_list,
+            init_weights=self.init_weights,
+            customized_test_config=self.customized_pose_config,
+        )
+
+        deeplabcut.create_labeled_video(
+            "",
+            [self.video_path],
+            videotype=self.videotype,
+            filtered=False,
+            init_weights=self.init_weights,
+            draw_skeleton=True,
+            superanimal_name=self.supermodel_name,
+            pcutoff=self.pcutoff,
+        )
```

Comment:
Is that still something we need since we no longer write images to disk?

Reply:
The create_labeled_video call under before_adapt is just to showcase what's the difference between using or without using video adaptation. It's not necessary but might be informative.

Reply:
I don't think that should be the default behavior: that'd be very costly on a long video, and I do not believe users would run the function blindly. You'd need to inspect the video first to determine whether adaptation is needed, don't you think?

Reply:
I agree.. But "inspecting video" is exactly what create_labeled_video under before_adapt is doing.